src/device.ts

```diff
@@ -308,24 +308,26 @@ export class Device extends vscode.Disposable {
 
     /**
      * Recursively create a directory (equivalent of mkdir -p).
-     * @param path the path of the directory
+     * @param filePath the path of the directory
      */
-    public async mkdir_p(path: string): Promise<void> {
-        const names = path.split('/');
-
-        // If path had a preceding slash, remove the empty path segment
-        if(!names[0]) {
-            names.shift();
+    public async mkdir_p(filePath: string): Promise<void> {
+        if (!path.posix.isAbsolute(filePath)) {
+            throw new URIError("The supplied file path must be absolute.");
```

Comment:
Why throw a `URIErrror` when there are no URIs here?

I don't have a feel for idiomatic node.js/typescript, but maybe an `AssertionError` would be more appropriate if you want something more specific than just `Error`?

Reply:
Or maybe use [assert](https://nodejs.org/api/assert.html)?

Reply:
JavaScript has very few useful error types, so most of the time it's just `Error` or a custom subclass. In this case I think I'll just replace it with a normal `Error`.

> Or maybe use assert?

Assertions really haven't caught on in Node.js... that doesn't mean that we couldn't pick up the tradition, but I don't see it used very often.